docs/dev_guides/changing_splink/testing.md

```diff
@@ -0,0 +1,283 @@
+
+The available options are:
+
+* `pytest tests/ -m core` - run only the 'core' tests, meaning those without dialect-dependence. In practice this means any test that hasn't been decorated using `mark_with_dialects_excluding` or `mark_with_dialects_including`.
+* `pytest tests/ -m duckdb` - run all `duckdb` tests, and all `core` tests
+    * & similarly for other dialects
+* `pytest tests/ -m duckdb_only` - run all `duckdb` tests only, and _not_ the `core` tests
+    * & similarly for other dialects
+* `pytest tests/ -m default` or equivalently `pytest tests/` - run all tests in the `default` group. The `default` group consists of the `core` tests, and those dialects in the `default` group - currently `spark` and `duckdb`.
+    * Other groups of dialects can be added and will similarly run with `pytest tests/ -m new_dialect_group`. Dialects within the current scope of testing and the groups they belong to are defined in the `dialect_groups` dictionary in [tests/decorator.py](https://github.com/moj-analytical-services/splink/blob/master/tests/decorator.py)
+* `pytest tests/ -m all` run all tests for all available dialects
+
+These all work alongside all the other pytest options, so for instance to run the tests for training `probability_two_random_records_match` for only `duckdb`, ignoring warnings, with quiet output, and exiting on the first failure/error:
+```sh
+pytest -W ignore -q -x -m duckdb tests/test_estimate_prob_two_rr_match.py
+```
+
+### Running tests with docker 🐳
+
+If you want to test Splink against a specific version of python, the easiest method is to utilise docker 🐳.
+
+Docker allows you to more quickly and easily install a specific version of python and run the existing test library against it.
+
+This is particularly useful if you're using py > 3.9.10 (which is currently in use in our tests github action) and need to run a secondary set of tests.
+
+A pre-built Dockerfile for running tests against python version 3.9.10 can be located within [scripts/run_tests.Dockerfile](https://github.com/moj-analytical-services/splink/blob/master/scripts/run_tests.Dockerfile).
+
+To run, simply use the following docker command from within a terminal and the root folder of a splink clone:
+```sh
+docker build -t run_tests:testing -f scripts/run_tests.Dockerfile . && docker run --rm --name splink-test run_tests:testing
+```
+
+This will both build and run the tests library.
+
+Feel free to replace `run_tests:testing` with an image name and tag you're happy with.
+
+Reusing the same image and tag will overwrite your existing image.
+
+You can also overwrite the default `CMD` if you want a different set of `pytest` command-line options, for example
+```sh
+docker run --rm --name splink-test run_tests:testing pytest -W ignore -m spark tests/test_u_train.py
+```
```

Comment:
Similar to other sections - I don't think this will apply to the majority of people, So I would be tempted to wrap up in a [collapsible block](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks)